extensions/cache/deployment/src/test/java/io/quarkus/cache/test/deployment/DeploymentExceptionsTest.java

```diff
@@ -0,0 +1,123 @@
+package io.quarkus.cache.test.deployment;
+
+import static java.util.Arrays.stream;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.stream.Stream;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.spi.DeploymentException;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.cache.Cache;
+import io.quarkus.cache.CacheInvalidate;
+import io.quarkus.cache.CacheInvalidateAll;
+import io.quarkus.cache.CacheName;
+import io.quarkus.cache.CacheResult;
+import io.quarkus.cache.deployment.exception.ClassTargetException;
+import io.quarkus.cache.deployment.exception.PrivateMethodTargetException;
+import io.quarkus.cache.deployment.exception.UnknownCacheNameException;
+import io.quarkus.cache.deployment.exception.VoidReturnTypeTargetException;
+import io.quarkus.test.QuarkusUnitTest;
+
+/**
+ * This class tests many kinds of {@link DeploymentException} causes related to caching annotations.
+ */
+public class DeploymentExceptionsTest {
+
+    private static final String UNKNOWN_CACHE_1 = "unknown-cache-1";
+    private static final String UNKNOWN_CACHE_2 = "unknown-cache-2";
+    private static final String UNKNOWN_CACHE_3 = "unknown-cache-3";
+
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class).addClasses(TestResource.class, TestBean.class))
+            .assertException(t -> {
+                assertEquals(DeploymentException.class, t.getClass());
+                assertEquals(10, t.getSuppressed().length);
+                assertPrivateMethodTargetException(t, "shouldThrowPrivateMethodTargetException", 1);
+                assertPrivateMethodTargetException(t, "shouldAlsoThrowPrivateMethodTargetException", 2);
+                assertVoidReturnTypeTargetException(t, "showThrowVoidReturnTypeTargetException");
+                assertClassTargetException(t, TestResource.class, 1);
+                assertClassTargetException(t, TestBean.class, 2);
+                assertUnknownCacheNameException(t, UNKNOWN_CACHE_1);
+                assertUnknownCacheNameException(t, UNKNOWN_CACHE_2);
+                assertUnknownCacheNameException(t, UNKNOWN_CACHE_3);
+            });
```

Comment:
Here's a preview of what would be logged if the exceptions were not "caught" by the test:
```
Caused by: javax.enterprise.inject.spi.DeploymentException: Found 10 deployment problems:
[1] @CacheResult is not allowed on a method returning void [class=io.quarkus.cache.test.deployment.DeploymentExceptionsTest$TestResource, method=showThrowVoidReturnTypeTargetException]
[2] Caching annotations are not allowed on a class [class=io.quarkus.cache.test.deployment.DeploymentExceptionsTest$TestResource, annotation=io.quarkus.cache.CacheInvalidate]
[3] Caching annotations are not allowed on a private method [class=io.quarkus.cache.test.deployment.DeploymentExceptionsTest$TestResource, method=shouldThrowPrivateMethodTargetException, annotation=io.quarkus.cache.CacheInvalidateAll]
[4] Caching annotations are not allowed on a private method [class=io.quarkus.cache.test.deployment.DeploymentExceptionsTest$TestResource, method=shouldAlsoThrowPrivateMethodTargetException, annotation=io.quarkus.cache.CacheInvalidate]
[5] Caching annotations are not allowed on a private method [class=io.quarkus.cache.test.deployment.DeploymentExceptionsTest$TestResource, method=shouldAlsoThrowPrivateMethodTargetException, annotation=io.quarkus.cache.CacheInvalidate]
[6] Caching annotations are not allowed on a class [class=io.quarkus.cache.test.deployment.DeploymentExceptionsTest$TestBean, annotation=io.quarkus.cache.CacheInvalidateAll]
[7] Caching annotations are not allowed on a class [class=io.quarkus.cache.test.deployment.DeploymentExceptionsTest$TestBean, annotation=io.quarkus.cache.CacheInvalidateAll]
[8] A field or method parameter is annotated with a @CacheName annotation referencing an unknown cache name [class=io.quarkus.cache.test.deployment.DeploymentExceptionsTest$TestBean, cacheName=unknown-cache-1]
[9] A field or method parameter is annotated with a @CacheName annotation referencing an unknown cache name [class=io.quarkus.cache.test.deployment.DeploymentExceptionsTest$TestBean, cacheName=unknown-cache-2]
[10] A field or method parameter is annotated with a @CacheName annotation referencing an unknown cache name [class=io.quarkus.cache.test.deployment.DeploymentExceptionsTest$TestBean, cacheName=unknown-cache-3]
```